File: storage/db-core/src/main/java/com/coffee/gu/order/OrderLineEntity.java
package com.coffee.gu.order;

import com.coffee.gu.BaseEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;

import java.math.BigDecimal;

@Entity
@Table(name = "order_line")
public class OrderLineEntity extends BaseEntity {
    private String orderKey;
    private Long menuId;
    private String menuName;
    private String imageUrl;
    private String description;
    private Long quantity;
    private BigDecimal unitPrice;
    private BigDecimal totalPrice;
    private Boolean isStampEligible;

    public OrderLineEntity() {}

    public OrderLineEntity(String orderKey, Long menuId, String menuName, String imageUrl, String description, Long quantity, BigDecimal unitPrice, BigDecimal totalPrice, Boolean isStampEligible) {
        this.orderKey = orderKey;
        this.menuId = menuId;
        this.menuName = menuName;
        this.imageUrl = imageUrl;
        this.description = description;
        this.quantity = quantity;
        this.unitPrice = unitPrice;
        this.totalPrice = totalPrice;
        this.isStampEligible = isStampEligible;
    }

    public static OrderLineEntity from(OrderLine orderLine) {
        return new OrderLineEntity(
                orderLine.getOrderKey(),
                orderLine.getMenuId(),
                orderLine.getMenuName(),
                orderLine.getImageUrl(),
                orderLine.getDescription(),
                orderLine.getQuantity(),
                orderLine.getUnitPrice(),
                orderLine.getTotalPrice(),
                orderLine.getIsStampEligible()
        );
    }

    public OrderLine toModel() {
        return new OrderLine(
                this.id,
                this.orderKey,
                this.menuId,
                this.menuName,
                this.imageUrl,
                this.description,
                this.quantity,
                this.unitPrice,
                this.totalPrice,
                this.isStampEligible
        );
    }

    public String getOrderKey() {
        return orderKey;
    }

    public Long getMenuId() {
        return menuId;
    }

    public String getMenuName() {
        return menuName;
    }

    public String getImageUrl() {
        return imageUrl;
    }

    public Long getQuantity() {
        return quantity;
    }

    public BigDecimal getUnitPrice() {
        return unitPrice;
    }

    public BigDecimal getTotalPrice() {
        return totalPrice;
    }

    public String getDescription() {
        return description;
    }
}

File: storage/db-core/src/main/kotlin/com/coffee/gu/order/OrderLineEntity.kt
package com.coffee.gu.order

import com.coffee.gu.BaseEntity
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.math.BigDecimal

@Entity
@Table(name = "order_line")
class OrderLineEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0,
    val orderKey: String,
    val menuId: Long,
    val menuName: String,
    val imageUrl: String? = null,
    val description: String? = null,
    val quantity: Long,
    val unitPrice: BigDecimal,
    val totalPrice: BigDecimal,
    val isStampEligible: Boolean,
) : BaseEntity() {

    fun toModel(): OrderLine {
        return OrderLine(
            id = id,
            orderKey = orderKey,
            menuId = menuId,
            menuName = menuName,
            imageUrl = imageUrl,
            description = description,
            quantity = quantity,
            unitPrice = unitPrice,
            totalPrice = totalPrice,
            isStampEligible = isStampEligible,
        )
    }

    companion object {
        fun from(orderLine: OrderLine): OrderLineEntity = OrderLineEntity(
            id = orderLine.id ?: 0L,
            orderKey = orderLine.orderKey,
            menuId = orderLine.menuId,
            menuName = orderLine.menuName,
            imageUrl = orderLine.imageUrl,
            description = orderLine.description,
            quantity = orderLine.quantity,
            unitPrice = orderLine.unitPrice,
            totalPrice = orderLine.totalPrice,
            isStampEligible = orderLine.isStampEligible,
        )
    }
}
